plugins/scaffolder/src/api.ts

```diff
@@ -235,4 +241,15 @@ export class ScaffolderClient implements ScaffolderApi {
     const response = await fetch(`${baseUrl}/v2/actions`);
     return await response.json();
   }
+
+  registerField(name: string, field: Field): void {
```

Comment:
Not feeling that an API is best suited for this kind of component registration tbh. It may be that either a dedicated type of component extension (`createScaffolderFormExtension`) or just a prop somewhere is what we should be going for instead, but not too sure tbh.

Reply:
This makes sense to me since this custom component logic isn't related to the API but I can't understand _the how_ in creating a component extension or a prop, right now there's nothing to access a component outside of scaffolder besides the API/Client and `ScaffolderPage`.

Reply:
I can help out with getting it in place if we go down the route of component extensions, but kinda wanna settle if it's the API we want first.

It could look something like this:

```tsx
import { createScaffolderFormExtension } from '@backstage/plugin-scaffolder';

export const AutocompleteField = autocompletePlugin.provide(createScaffolderFormExtension({
  name: 'AutocompleteField',
  component: {
    async: () => import('./components/AutocompleteField').then(m => m.AutocompleteField)
  },
  validate: (...) => /* custom validation logic */,
})
```

To be used in the app like this:

```tsx
<FlatRoutes>
  ...
  <ScaffolderPage>
    <AutocompleteField />
  </ScaffolderPage>
</FlatRoutes>
```

Or maybe a but more explicit?

```tsx
<FlatRoutes>
  ...
  <ScaffolderPage>
    <ScaffolderPage.FormFields>
      <AutocompleteField />
    </ScaffolderPage.FormFields>
  </ScaffolderPage>
</FlatRoutes>
```

Thoughts on ^ @backstage/maintainers ?

Reply:
uhm I think this is not what I'm trying to get done, I as a user of scaffolder I do not have access to modify the ScaffolderPage component, what I want is a way to register my custom components to rjsf this way any template using a custom field will be rendered by the custom field, like in the documentation from rjsf.

https://react-jsonschema-form.readthedocs.io/en/docs/advanced-customization/custom-widgets-fields/#custom-field-components

TLDR: I want to add my components in `packages/app`

Reply:
It totally is. You wouldn't be modifying the `ScaffolderPage` component, just supplying some extra children to it in the app, which are then interpreted as components to add as rjsf fields.

Reply:
you are right, I can see it now! 👁👁, your approach looks more advanced and decoupled from my initial implementation, gonna give it a try, thanks!

Reply:
I like the more explicit approach and wrapping them in the `ScaffolderPage.FormFields` section.

Gives more flexibility to the JSX API too in case we start including other extension points (Which is always most probably the case).

Reply:
@Rugvip Well I've been trying to wrap my head around converting this into an extension but I haven't found any documentation on how to implement it, I found [this](https://backstage.io/docs/plugins/composability#extensions-from-a-plugins-point-of-view) but is about how to use it, is there any reference I can look at?